extensions/mailer/runtime/src/main/java/io/quarkus/mailer/reactive/ReactiveMailer.java

```diff
@@ -0,0 +1,19 @@
+package io.quarkus.mailer.reactive;
+
+import io.quarkus.mailer.Mail;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * A mailer to send email asynchronously.
+ */
+public interface ReactiveMailer {
+
+    /**
+     * Sends the given emails.
+     *
+     * @param mails the emails to send, must not be {@code null}, must not contain {@code null}
+     * @return an {@link Uni} indicating when the mails have been sent. The {@link Uni} may fire a failure if the
+     *         emails cannot be sent.
+     */
+    Uni<Void> send(Mail... mails);
```

Comment:
Perhaps the returned object may give more info instead of `Void``? `MailResult` perhaps?

Reply:
And what would it contains? Basically it just tells you success or failure which is conveyed using the Uni events.

Reply:
I was thinking about the message itself but maybe `Uni` already handles that